packages/editor/src/client/blocks/richTextBlock/buttons.tsx

```diff
@@ -0,0 +1,234 @@
+import React, {useState} from 'react'
+import {useTranslation} from 'react-i18next'
+import {Button, ControlLabel, Form, FormControl, FormGroup, Modal} from 'rsuite'
+import {Editor, Transforms, Range} from 'slate'
+import {useSlate} from 'slate-react'
+import {
+  ToolbarIconButtonProps,
+  ToolbarButtonProps,
+  ToolbarButton,
+  ToolbarIconButton
+} from '../../atoms/toolbar'
+import {isFormatActive, toggleFormat} from './editorUtils'
+import {Format, InlineFormat} from './formats'
+
+interface SlateBlockButtonProps extends ToolbarIconButtonProps {
+  readonly format: Format
+}
+
+export function FormatIconButton({icon, format}: SlateBlockButtonProps) {
+  const editor = useSlate()
+
+  return (
+    <ToolbarIconButton
+      icon={icon}
+      active={isFormatActive(editor, format)}
+      onMouseDown={e => {
+        e.preventDefault()
+        toggleFormat(editor, format)
+      }}
+    />
+  )
+}
+
+interface SlateBlockButtonWithChildrenProps extends ToolbarButtonProps {
+  readonly format: Format
+}
+
+export function FormatButtonWithChildren({format, children}: SlateBlockButtonWithChildrenProps) {
+  const editor = useSlate()
+
+  return (
+    <ToolbarButton
+      active={isFormatActive(editor, format)}
+      onMouseDown={e => {
+        e.preventDefault()
+        toggleFormat(editor, format)
+      }}>
+      {children}
+    </ToolbarButton>
+  )
+}
+
+export function LinkFormatButton() {
```

Comment:
Why is this here? It looks like the LinkFormat buttons could be also in their own file.

Reply:
agree